tasks/lib/code-coverage-enforcer-lib.js

```diff
@@ -121,13 +121,14 @@ module.exports = (function () {
      * @param {array}  files   a pointer to an array that stores a list of files
      * @param {array}  includes a list of patterns for files to match
      * @param {array}  excludes a list of patterns for files to ignore
+     * @param {string} extension    an optional regular expression for matching file extensions
      *
      * collect(options.src, fileList, options.includes, options.excludes, null);
      */
-    exports.collect = function (fp, files, includes, excludes, replaceDirectory) {
+    exports.collect = function (fp, files, includes, excludes, replaceDirectory, extension) {
         if (shjs.test("-d", fp)) {
             shjs.find(fp).filter(function (file) {
-                if (file.match(/\.js$/)) {
+                if (file.match(extension || /\.js$/)) {
```

Comment:
It sort of depends whether our goal should be just to expand the functionality to consider .jsx files, or for the user to be able to specify arbitrary file extensions.  I was going for the latter, but the former would meet our team's requirements as well.

So in the code as written, we use the user specified extension regex for the match if one has been specified.  Otherwise, we use /.js$/ as a default.